dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/tcp/AccessLists.java

```diff
@@ -0,0 +1,195 @@
+     */
+    public AccessLists(FilterList _excludeAccess, FilterListStr _excludeAccessNames, FilterList _includeAccess,
+                       FilterListStr _includeAccessNames, boolean _caseInsensitiveHostnames) {
+        this.excludeAccess = _excludeAccess;
+        this.includeAccess = _includeAccess;
+        this.excludeAccessNames = _excludeAccessNames;
+        this.includeAccessNames = _includeAccessNames;
+        this.caseInsensitiveHostnames = _caseInsensitiveHostnames;
+    }
+
+    protected static AccessLists getInstance(TCPConfigurationImpl config) {
+        AccessLists retVal = null;
+        boolean haveAList = false;
+        FilterList excludeAccess = null;
+        FilterList includeAccess = null;
+        FilterListStr excludeAccessNames = null;
+        FilterListStr includeAccessNames = null;
+
+        String[] sExclude = config.getAddressExcludeList();
+        excludeAccess = new FilterList();
+        if (sExclude != null) {
+            excludeAccess.buildData(sExclude, false);
+            excludeAccess.setActive(true);
+            haveAList = true;
+        }
+
+        sExclude = config.getHostNameExcludeList();
+        excludeAccessNames = new FilterListFastStr();
+        if (sExclude != null) {
+            if (excludeAccessNames.buildData(sExclude) == false) {
+                excludeAccessNames = new FilterListSlowStr();
+                excludeAccessNames.buildData(sExclude);
+            }
+            excludeAccessNames.setActive(true);
+            haveAList = true;
+        }
+
+        String[] sInclude = config.getAddressIncludeList();
+        includeAccess = new FilterList();
+        if (sInclude != null) {
+            includeAccess.buildData(sInclude, false);
+            includeAccess.setActive(true);
+            haveAList = true;
+        }
+
+        sInclude = config.getHostNameIncludeList();
+        includeAccessNames = new FilterListFastStr();
+        if (sInclude != null) {
+            if (includeAccessNames.buildData(sInclude) == false) {
+                includeAccessNames = new FilterListSlowStr();
+                includeAccessNames.buildData(sInclude);
+
+            }
+            includeAccessNames.setActive(true);
+            haveAList = true;
+        }
+
+        if (haveAList) {
+            retVal = new AccessLists(excludeAccess, excludeAccessNames, includeAccess, includeAccessNames,
+                            config.getCaseInsensitiveHostnames());
+        }
+
+        return retVal;
+    }
+
+    /**
+     * Query whether a given client address is denied by this configuration.
+     * 
+     * @param remoteAddr
+     * @return boolean
+     */
+    public boolean accessDenied(InetAddress remoteAddr) {
+
+        String hostname = null; //F184719
+
+        // check the inclusion lists first to see if the client matches
+        if (includeAccess.getActive() || includeAccessNames.getActive()) {
+            boolean closeSocket = true;
```

Comment:
I think this would be better to be called 'denied' to is obvious it is the result and the caller chooses what to do - but if not new code then perhaps left as is.

Reply:
This is taken directly from the channelfw tcpchannel - I agree it would be better if we didn't have to accept the connection at all, but when this code is run the socket has been opened and we do need to close it.

Reply:
either way we'll be changing this up later with https://github.com/OpenLiberty/open-liberty/issues/18438